zulipterminal/ui_tools/utils.py

```diff
@@ -18,8 +21,47 @@ def create_msg_box_list(model: Any, messages: Union[None, Iterable[Any]]=None,
     message_list.sort(key=lambda msg: msg['timestamp'])
     w_list = []
     focus_msg = None
+    is_empty_narrow = False
     last_msg = last_message
     muted_msgs = 0  # No of messages that are muted.
+    if last_msg is not None and last_msg['id'] is None:
+        last_message = None
+    # We create a dummy message to show in narrows with
+    # no previous messages, whose author is welcome bot.
+    if (message_list == [] and (stream_caption is not None or
+                                pm_details is not None)):
+        msg = {}
```

Comment:
To clarify, rather than
```python3
x = {}
if y:
    x.update(dict1)
elif z:
    x.update(dict2)
```
it may be clearer to have:
```python3
if y:
    x = dict1
elif z:
    x = dict2
```